packages/@lwc/ssr-compiler/src/compile-js/generate-markup.ts

```diff
@@ -87,10 +89,17 @@ const bAssignGenerateMarkupToComponentClass = esTemplate`
 export function addGenerateMarkupExport(
     program: Program,
     state: ComponentMetaState,
+    options: TransformOptions,
     filename: string
 ) {
     const { hasRenderMethod, privateFields, publicFields, tmplExplicitImports } = state;
+    const { namespace, name } = options;
 
+    // The default tag name represents the component name that's passed to the transformer.
+    // This is needed to generate markup for dynamic components which are invoked through
+    // the generateMarkup function on the constructor.
+    // At the time of generation, the invoker does not have reference to its tag name to pass as an argument.
+    const defaultTagName = b.literal(`${namespace}-${name}`);
```

Comment:
Just double checking: is the `name` value here kebab-cased or camel-cased?

Reply:
Added a helper [function](https://github.com/salesforce/lwc/pull/4847/files#diff-3c40f7d063edbf5c496846dbd7e46d2c3b792e9be53fa45770b2c14636b053a6R17-R20)!